### PR TITLE
feat(split-data-loader): add babel

### PR DESCRIPTION
**Pull request by rohanmurthy**
This PR adds babel to this repository. Since we want `lib` to contain only our transpiled javascript code, I moved all of the `lib` files to `src`. Also, per @fritzmonkey 's suggestion, I changed all the `require`s to `import`s.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,9 @@
   "description": "",
   "main": "lib/index.js",
   "scripts": {
+    "build": "babel lib -d lib --ignore 'lib/*.test.js'",
+    "build-dry": "babel lib --ignore 'lib/*.test.js'",
+    "prepublishOnly": "npm run build",
```
Comment (rohanmurthy) [inline package.json]: We only want to transpile our code before we publish this package.

File: package.json [hunk 2]
```diff
@@ -4,6 +4,9 @@
   "description": "",
   "main": "lib/index.js",
   "scripts": {
+    "build": "babel lib -d lib --ignore 'lib/*.test.js'",
+    "build-dry": "babel lib --ignore 'lib/*.test.js'",
```
Comment (silasbw) [inline package.json]: update CI to run `build-dry` (i.e., test the transpilation).

File: package.json [hunk 3]
```diff
@@ -4,6 +4,9 @@
   "description": "",
   "main": "lib/index.js",
   "scripts": {
+    "build": "babel lib -d lib --ignore 'lib/*.test.js'",
```
Comment (fritzmonkey) [inline package.json]: Just to confirm, this is babeling everything (excluding tests) in `lib` and outputting it back to `lib`. Is this the intention? The pattern that I've seen more often is to have the non-transpiled code live in `src` and the transpiled output live in `lib`.

Comment (silasbw) [reply package.json]: ah, let's follow the convention @fritzmonkey points out.

**Pull request merged**